bundles/org.openhab.binding.smsmodem/src/main/java/org/openhab/binding/smsmodem/internal/handler/SMSModemBridgeHandler.java

```diff
@@ -0,0 +1,473 @@
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.smsmodem.internal.SMSConversationDiscoveryService;
+import org.openhab.binding.smsmodem.internal.SMSModemBindingConstants;
+import org.openhab.binding.smsmodem.internal.SMSModemBridgeConfiguration;
+import org.openhab.binding.smsmodem.internal.SMSModemRemoteBridgeConfiguration;
+import org.openhab.binding.smsmodem.internal.actions.SMSModemActions;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.io.transport.serial.SerialPortIdentifier;
+import org.openhab.core.io.transport.serial.SerialPortManager;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.smslib.CommunicationException;
+import org.smslib.Modem;
+import org.smslib.Modem.Status;
+import org.smslib.callback.IDeviceInformationListener;
+import org.smslib.callback.IInboundOutboundMessageListener;
+import org.smslib.callback.IModemStatusListener;
+import org.smslib.message.AbstractMessage.Encoding;
+import org.smslib.message.DeliveryReportMessage;
+import org.smslib.message.InboundMessage;
+import org.smslib.message.MsIsdn;
+import org.smslib.message.OutboundMessage;
+import org.smslib.message.Payload;
+import org.smslib.message.Payload.Type;
+
+/**
+ * The {@link SMSModemBridgeHandler} is responsible for handling
+ * communication with the modem.
+ *
+ * @author Gwendal ROULLEAU - Initial contribution
+ */
+@NonNullByDefault
+public class SMSModemBridgeHandler extends BaseBridgeHandler
+        implements IModemStatusListener, IInboundOutboundMessageListener, IDeviceInformationListener {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Set.of(
+            SMSModemBindingConstants.SMSMODEMBRIDGE_THING_TYPE,
+            SMSModemBindingConstants.SMSMODEMREMOTEBRIDGE_THING_TYPE);
+
+    private final Logger logger = LoggerFactory.getLogger(SMSModemBridgeHandler.class);
+
+    private Set<SMSConversationHandler> childHandlers = new HashSet<>();
+
+    private SerialPortManager serialPortManager;
+
+    /**
+     * The smslib object responsible for the serial communication with the modem
+     */
+    private @NonNullByDefault({}) Modem modem;
+
+    /**
+     * A scheduled watchdog check
+     */
+    private @NonNullByDefault({}) ScheduledFuture<?> checkScheduled;
```

Comment:
The null check annotations should only be disabled in certain cases like injection. These `@NonNullByDefault({})` should be removed and the attributes be checked for null values before they are accessed.

Reply:
Done